pyproject.toml

```diff
@@ -57,6 +57,8 @@ simulators = [
     "cython",
     "pybind11>=2.2.3,<3.0",
     "projectq>=0.5.0,<0.9.0",
+    "cuquantum-python>=23.6.0",
+    "cupy>=10.4.0",
```

Comment:
Looks like CI is no longer pleased with me requiring CuPy as an optional dependency:

>       ERROR: CUDA could not be found on your system.
>       
>       HINT: You are trying to build CuPy from source, which is NOT recommended for general use.
>             Please consider using binary packages instead.

Any suggestions on what to do? I'm happy to just remove CuPy and cuquantum-python from the optional dependencies. To install these you need a machine with a GPU, which I don't expect is the case for the machine running CI?

Reply:
@PabloAndresCQ You can create a new dependency section named `gpu` or `cuda` (and refer it in README and elsewhere for this sim), like so:
```
gpu = [
    "cuquantum-python>=23.6.0",
    "cupy>=10.4.0",
]
```

Since `simulators` in included in `all` below, the CI step at https://github.com/PECOS-packages/PECOS/blob/development/.github/workflows/python-app.yml#L41-L45 is including these dependencies in it.

Reply:
Awesome, thanks! :)